clientd3d/d3drender.c

```diff
@@ -29,11 +33,33 @@ inline float FovVertical(long height)
 	return height / (float)(main_viewport_height) * (PI / 5.6f);
 }
 
-// Calculates the intensity of an animation based on the frame number.
+// Get the number of milliseconds lapsed since the epoch.
+int getMillisecondsLapsed()
+{
+	auto time_point = chrono_time_now();
+	auto duration_since_epoch = time_point.time_since_epoch();
+	return std::chrono::duration_cast<std::chrono::milliseconds>(duration_since_epoch).count();
+}
+
+// Calculates the intensity of an animation based on the time.
 // This can be used to cycle in a pattern e.g. for invisibility.
-float animationIntensity(int frameNumber)
+float animationIntensity(int milliseconds)
+{	
+	return (milliseconds & 3) / 256.0f;
+}
+
+// Calculates the intensity of an animation based on the time.
+void updateRenderChunkAnimationIntensity(d3d_render_chunk_new* pChunk)
 {
-	return (frameNumber & 3) / 256.0f;
+	int milliseconds = getMillisecondsLapsed();
+	pChunk->st1[0].s -= animationIntensity(milliseconds);

```

Comment:
Call animationIntensity once and reuse the value.  And then, once you're doing that, you probably no longer need animationIntensity as a separate function.